vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/processors/behavioral/bandits/epsilon_greedy_bandit_processor.cc

```diff
@@ -0,0 +1,151 @@
+    case AdNotificationEventType::kTimedOut:
+    case AdNotificationEventType::kDismissed: {
+      UpdateArm(/* reward */ 0, segment);
+      break;
+    }
+    case AdNotificationEventType::kClicked: {
+      UpdateArm(/* reward */ 1, segment);
+      break;
+    }
+    case AdNotificationEventType::kViewed: {
+      NOTREACHED();
+      break;
+    }
+  }
+
+  BLOG(1, "Processed ad event");
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+EpsilonGreedyBanditArmMap EpsilonGreedyBandit::MaybeAddOrResetArms(
+    const EpsilonGreedyBanditArmMap& arms) const {
+  EpsilonGreedyBanditArmMap updated_arms = arms;
+  for (const auto& segment : resource::kSegments) {
+    auto iter = updated_arms.find(segment);
+
+    if (iter == updated_arms.end()) {
+      EpsilonGreedyBanditArmInfo arm;
+      arm.value = kArmDefaultValue;
+      arm.pulls = kArmDefaultPulls;
+      updated_arms[segment] = arm;
+      BLOG(6, "Added arm for segment " << segment);
+      continue;
+    }
+
+    if (!iter->second.IsValid()) {
+      EpsilonGreedyBanditArmInfo new_arm;
+      new_arm.value = kArmDefaultValue;
+      new_arm.pulls = kArmDefaultPulls;
+      updated_arms[segment] = new_arm;
+      BLOG(6, "Did reset invalid arm for segment  " << segment);
+      continue;
+    }
+
+    BLOG(6, "Arm exists for segment " << segment);
+  }
+
+  return updated_arms;
+}
+
+EpsilonGreedyBanditArmMap EpsilonGreedyBandit::MaybeDeleteArms(
+    const EpsilonGreedyBanditArmMap& arms) const {
+  EpsilonGreedyBanditArmMap updated_arms = arms;
+  for (const auto& arm : updated_arms) {
+    auto iter = std::find(resource::kSegments.begin(),
+        resource::kSegments.end(), arm.first);
+
+    if (iter == resource::kSegments.end()) {
+      updated_arms.erase(arm.first);
+      BLOG(6, "Deleted arm for segment " << arm.first);
+    }
+  }
+
+  return updated_arms;
+}
+
+void EpsilonGreedyBandit::InitializeArms() const {
+  const std::string json = AdsClientHelper::Get()->GetStringPref(
+      prefs::kEpsilonGreedyBanditArms);
+  EpsilonGreedyBanditArmMap arms = EpsilonGreedyBanditArms::FromJson(json);
+
+  arms = MaybeAddOrResetArms(arms);
+  arms = MaybeDeleteArms(arms);
+
+  const std::string json_out = EpsilonGreedyBanditArms::ToJson(arms);
+  AdsClientHelper::Get()->SetStringPref(
+      prefs::kEpsilonGreedyBanditArms, json_out);
+  BLOG(1, "Successfully initialized arms");
+}
+
+void EpsilonGreedyBandit::UpdateArm(
+    uint64_t reward,
+    const std::string& segment) const {
+  const std::string json = AdsClientHelper::Get()->GetStringPref(
+      prefs::kEpsilonGreedyBanditArms);
+  EpsilonGreedyBanditArmMap arms = EpsilonGreedyBanditArms::FromJson(json);
+
+  if (arms.empty()) {
+    return;
+  }
+
+  auto iter = arms.find(segment);
+  if (iter == arms.end()) {
+    BLOG(1, "Arm not found for segment " << segment);
+    return;
+  }
+
+  EpsilonGreedyBanditArmInfo arm = iter->second;
+  arm.pulls = arm.pulls + 1;
```

Comment:
nit: `arms.pulls++`

Reply:
this results in tests failing - let's discuss